sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -31,14 +33,28 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
    * @param locationId location to be updated.
    * @returns persisted location.
    */
-  async saveLocation(
+  async createLocation(
     institutionId: number,
     data: InstitutionLocationModel,
     auditUserId: number,
     locationId?: number,
   ): Promise<InstitutionLocation> {
     const auditUser = { id: auditUserId } as User;
     const institution = { id: institutionId };
+
+    const isInstitutionCodeDuplicate = await this.hasLocationCodeForInstitution(
+      data.institutionCode,
+      locationId,
```

Comment:
I believe that these parameters (locationId, institutionId) are mutually exclusive. I would recommend providing one or another. I would recommend keeping the institution one that makes the query more simple.

Reply:
ok! Done. Thanks!